### PR TITLE
Raise ValueError for incorrect array shapes in archive methods

### PR DESCRIPTION
**Pull request by btjanaka**
## TODO

- [x] Add checking utils to all methods
  - [x] index_of / index_of_single
  - [x] elites_with_measures / elites_with_measures_single
  - [x] int_to_grid_index / grid_to_int_index
- [x] Test that errors are raised
- [x] Fix any errors we discover along the way

## Status

- [x] I have read the guidelines in [CONTRIBUTING.md](https://github.com/icaros-usc/pyribs/blob/master/CONTRIBUTING.md).
- [x] I have formatted my code using `yapf`
- [x] I have tested my code by running `pytest`
- [x] I have linted my code with `pylint`
- [x] I have added a one-line description of my change to the changelog in `HISTORY.md`.
- [x] This PR is ready to go

### CODE REVIEW

File: ribs/_utils.py [hunk 1]
```diff
@@ -0,0 +1,27 @@
+"""Miscellaneous internal utilities."""
+
+
+def check_measures_batch_shape(measures_batch,
+                               measure_dim,
+                               name="measures_batch"):
```
Comment (itsdawei) [inline ribs/_utils.py:6]: You have a `name` parameter here but not for the `check_measures_shape` function

**Review by itsdawei — APPROVED** [review]
LGTM

File: ribs/_utils.py (hunk 1 above)
Comment (btjanaka) [reply ribs/_utils.py:6]: Added name to `check_measures_shape`

File: ribs/archives/_grid_archive.py [hunk 2]
```diff
@@ -195,9 +210,15 @@ def int_to_grid_index(self, int_index_batch):
...
+        Raises:
+            ValueError: ``int_index_batch`` is not of shape (batch_size,).
         """
-        return np.asarray(
-            np.unravel_index(
-                np.asarray(int_index_batch),
-                self._dims,
-            )).T.astype(np.int32)
+        int_index_batch = np.asarray(int_index_batch)
+        if len(int_index_batch.shape) != 1:
```
[the remark this answers is not in the document]
Comment (btjanaka) [reply ribs/archives/_grid_archive.py:217]: we only use it once, so not much point in doing so

**Pull request merged**